src/Symfony/Component/ClassLoader/ClassCollectionLoader.php

```diff
@@ -117,8 +117,17 @@ public static function load($classes, $cacheDir, $name, $autoReload, $adaptive =
 
         // cache the core classes
         if (!is_dir(dirname($cache))) {
-            mkdir(dirname($cache), 0777, true);
+            if (true !== @mkdir($cache, 0777, true)) {
+                $error = error_get_last();
+                if (!is_dir($cache)) {
+                    if ($error) {
+                        throw new \RuntimeException(sprintf('Failed to create "%s": %s.', $cache, $error['message']));
```

Comment:
Thank you for the suggestion. There are few more cases where `mkdir()` can fail, some do not directly apply to this part of the code. However `No space left on device` is one I encountered a few times (especially on embedded systems).

Assuming `error_get_last` will always return some (valid) is not something I would go for because it:
* is not guaranteed by design of PHP (ie. RT vendors can differ)
* users can change the behaviour
* not assuming that is set is not hard and the cost in both code size and performance is very small

So I tweaked your code a bit:

```php
<?php
$errors = array(
    array('message' => 'mkdir(): Permission denied'),
    array('message' => 'mkdir(): No space left on device'),
    array('message' => 'mkdir(): No such file or directory'),
    array('message' => 'mkdir() expects parameter 4 to be resource, null given'),
    array('message' => 'mkdir(): Invalid argument'), // when given an invalid directory name, like '/a"b/c'/
    array('message' => null),
    null,
);

$cache = '/cache/tmp/';
foreach($errors as $error) {
    $message = sprintf('mkdir(%s):%s', var_export($cache, true), isset($error['message']) ? preg_replace('/^(mkdir\(\))[:]?/', '', $error['message']) : ' Failed');
    echo $message."\n";
}

//mkdir('/cache/tmp/'): Permission denied
//mkdir('/cache/tmp/'): No space left on device
//mkdir('/cache/tmp/'): No such file or directory
//mkdir('/cache/tmp/'): expects parameter 4 to be resource, null given
//mkdir('/cache/tmp/'): Invalid argument
//mkdir('/cache/tmp/'): Failed
//mkdir('/cache/tmp/'): Failed
```

(on PHP 5.5.9-1ubuntu4.9 (cli) (built: Apr 17 2015 11:44:57) )